jep/0000/README.adoc

```diff
@@ -0,0 +1,124 @@
+* Code in Jenkins core (where necessary) switches behaviour between more secure yet non compliant cryptography to FIPS-140 compliant cryptography.
+* Non compliant and less secure crytpography has been replaced by a secure and compliant option.
+* General guidelines exist to help developers write FIPS-140 compliant code and/or understand incompatibilities with the standard.
+* User facing documentation provides a mere overview of the FIPS flag and what it does (and does not do).
+
+== Motivation
+
+Currently it is not possible to run Jenkins in a link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140-2] compliant way.  Many users needing Jenkins are required (e.g. for US government projects) to have any solution be compliant with FIPS-140 and today it is not possible for them to do this. 
+
+Both Jenkins core and plugins use cryptography that is not allowed and/or are not from an approved implementation, and this prevents adoption of Jenkins by these end users.
+Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall fowl of the standard due to its use of ``blowfish`` which is not an approved algorithm.
+
+== Reasoning
+
+In order to be FIPS-140 compliant the encryption algorithms used need to be validated, this requires complex configuration of the host, the JVM as well as the changes to Jenkins and plugin code as well as checking of any plugin code that would be installed.  As such it is not considered that the Jenkins community supports, or contains instructions on how to run Jenkins in a compliant manner, but merely contains only an overview to say it may be possible with various configuration of the JVM, Servlet environment, limitations of plugins and finally instructing Jenkins to prefer a FIPS-140 mode of operation.
+
+As of the time of writing the validated Java implementations are only FIPS-140-2 compliant even though FIPS-140-3 has superseded it.  As such the initial implementations will be targeting the older FIPS-140-2 standard.
+
+== Backwards Compatibility
+
+It is out of scope to support previous configuration, as it would not be possible to migrate non FIPS-140 compliant data to FIPS-140 compliant data within a FIPS-140 compliant environment.  As such there is no backwards compatibility with previous versions.
+
+It may but is not a requirement of this JEP be possible to switch from a FIPS-140 compliant environment to a non FIPS-140 compliant environment.
+
+Code changes will however be backwards compatible; if the system is running in a non FIPS-140 compliant environment then it will continue to run and operate the same after any changes due to this JEP.
+
+== Security
+
+Some of the cryptography supported in FIPS 140 (especially FIPS 140-2) is considered less secure than some modern alternatives.  There is no alternative in cases like these where FIPS-140 compliance is required.
```

Comment:
Optional rephrasing of the second sentence.  You're welcome to ignore this rephrasing as well.  The existing phrasing states the facts correctly.
```suggestion
Some of the cryptography supported in FIPS 140 (especially FIPS 140-2) is considered less secure than some modern alternatives.  FIPS 140-2 cryptography will be used when Jenkins is running in FIPS-140 mode even if that cryptography is considered less secure than other alternatives.
```